atc/db/resource_config_scope.go

```diff
@@ -330,7 +330,10 @@ func requestScheduleForJobsUsingResourceConfigScope(tx Tx, rcsID int) error {
 					"ji.passed_job_id":           nil,
 					"ji.trigger":                 false,
 				},
-				sq.NotEq{"j.next_build_id": nil},
+				sq.Or{
+					sq.NotEq{"j.next_build_id": nil},
+					sq.Eq{"j.inputs_determined": false},
```

Comment:
where is it checking `BuildInput.ResolveError is not empty` as you mentioned?

Reply:
1. `job.SaveNextInputMapping` will update `jobs.inputs_determined` with variable `resolved`
    -  https://github.com/concourse/concourse/blob/master/atc/scheduler/scheduler.go#L51 
2. For variable `resolved`, it determined by existence of resolve error
    - https://github.com/concourse/concourse/blob/7cff39e2086fcfb8c6d23f8b72fe8dc113eebab7/atc/scheduler/algorithm/compute.go#L61